src/app/auth/signup/signup.component.ts

```diff
@@ -0,0 +1,50 @@
+import {Component, OnInit} from '@angular/core';
+import {FormControl, FormGroupDirective, NgForm, Validators, FormBuilder, FormGroup} from '@angular/forms';
+import {ErrorStateMatcher} from '@angular/material/core';
+import { AuthService } from '../services/auth.service';
+import { Router } from '@angular/router';
+
+/** Error when invalid control is dirty, touched, or submitted. */
+export class MyErrorStateMatcher implements ErrorStateMatcher {
```

Comment:
@Luleherll it is not advisable to have more than one class in a single file, could you kindly consider placing this class in its own file